### PR TITLE
Make quorum unreachable if validator list expires (RIPD-1539)

### PR DESCRIPTION
**Pull request by wilsonianb**

### CODE REVIEW

File: src/ripple/app/misc/ValidatorList.h [hunk 1]
```diff
@@ -377,7 +377,8 @@ ValidatorList::onConsensusStart (
                 list.second.expiration <=
                 timeKeeper_.now().time_since_epoch().count())
             removePublisherList (list.first);
-        else if (! list.second.available)
```
Comment (bachase) [inline src/ripple/app/misc/ValidatorList.h]: Are all published lists required to have an expiration?

File: src/ripple/app/misc/impl/ValidatorList.cpp [hunk 2]
```diff
@@ -395,6 +395,9 @@ ValidatorList::removePublisherList (PublicKey const& publisherKey)
             --iVal->second;
     }
 
+    iList->second.list.clear();
```
Comment (bachase) [inline src/ripple/app/misc/impl/ValidatorList.cpp]: Without this call to `clear`, would that mean if a published list with keys `K1,K2` expired, then was republished with key `K1`, the `keyListsings_` count for `K2` would have been decremented twice? If so, is it worth adding a unit test to cover that case?

File: src/ripple/app/misc/ValidatorList.h (hunk 1 above)
Comment (wilsonianb) [reply src/ripple/app/misc/ValidatorList.h]: Yes
https://github.com/ripple/rippled/blob/3e5490ef6db9cbc3abdff04655d80952b288d097/src/ripple/app/misc/impl/ValidatorList.cpp#L302-L313

File: src/ripple/app/misc/impl/ValidatorList.cpp (hunk 2 above)
Comment (wilsonianb) [reply src/ripple/app/misc/impl/ValidatorList.cpp]: Good observation. This added test fails without the `clear`
https://github.com/ripple/rippled/pull/2240/commits/3bb20f990330afe60e043a53acacccc939c0a222

**Comment by bachase** [comment]
Great test, still 👍 on the PR.

**Comment by wilsonianb** [comment]
Squashed and rebased on 0.80.0

**Comment by bachase** [comment]
Merged as 02059a27d68a01054713dc9994d3af9b7724b76b

**Pull request closed without merging**